### PR TITLE
[Integration] Add support for syncing cloud resources from Datadog, Dynatrace, and New Relic

### PR DESCRIPTION
**Pull request by shariff-6**
# Description

## What

This PR adds support and examples for syncing cloud resources from Datadog, Dynatrace, and New Relic integrations.

## Why

 Encourage customers to leverage observability tools to ingest cloud resources into port

## How

- Added blueprints and entity mappings for Datadog, Dynatrace, and New Relic cloud resources.

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync finishes successfully
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Resync finishes successfully
- [ ] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [ ] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [ ] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: integrations/newrelic/.port/resources/port-app-config.yaml [hunk 1]
```diff
@@ -90,3 +90,32 @@ resources:
...
+        - AWS_S3_BUCKET
+        - AWS_RDS_DB_INSTANCE
+        - AWS_LAMBDA_FUNCTION
+        - AWS_ELB_LOAD_BALANCER
+        - AZURE_VIRTUAL_MACHINE
+        - AZURE_SQL_DATABASE
+        - GCP_COMPUTE_INSTANCE
+        - GCP_STORAGE_BUCKET
+        - GCP_SQL_DATABASE_INSTANCE
+        # Add more infrastructure integration types as needed
```
Comment (Tankilevitch) [inline integrations/newrelic/.port/resources/port-app-config.yaml:107]: based on this - https://github.com/port-labs/ocean/blob/7e7688363271ac49cc2f6c6ee8e48849494c714b/integrations/newrelic/newrelic_integration/overrides.py#L17
I don't see we have this kind of option in the mapping, was this tested?

Comment (Tankilevitch) [reply integrations/newrelic/.port/resources/port-app-config.yaml:107]: are you referring to `newrelic_types`?

File: integrations/dynatrace/.port/resources/port-app-config.yaml [hunk 2]
```diff
@@ -60,3 +60,34 @@ resources:
...
+        - "cloud:aws:acmprivateca"
+        - "cloud:aws:api_gateway"
+        - "cloud:aws:app_runner"
+        - "cloud:aws:appstream"
+        - "cloud:aws:appsync"
+        - "cloud:azure:apimanagement:service"
+        - "cloud:azure:app:containerapps"
+        - "cloud:azure:app:managedenvironments"
+        - "cloud:azure:appconfiguration:configurationstores"
+        - "cloud:azure:appplatform:spring"
```
Comment (Tankilevitch) [inline integrations/dynatrace/.port/resources/port-app-config.yaml:82]: was this tested?

Comment (shariff-6) [reply integrations/dynatrace/.port/resources/port-app-config.yaml:82]: Yes, this has been tested

File: integrations/newrelic/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 <!-- towncrier release notes start -->
 
+## 0.2.00 (2024-11-25)
```
Comment (Tankilevitch) [inline integrations/newrelic/CHANGELOG.md:10]:
```suggestion
## 0.1.100 (2024-11-25)
```

**Review by Tankilevitch — APPROVED** [review]
LGTM

**Pull request merged**